plugins/modules/registration_command.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) Evgeni Golov
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: registration_command
+version_added: 4.0.0
+short_description: Manage Registration Command
+description:
+  - Manage Registration Command
+author:
+  - "Evgeni Golov (@evgeni)"
+options:
+  activation_keys:
+    description:
+    - Activation keys for subscription-manager client, required for CentOS and Red Hat
+      Enterprise Linux.
+    - Required only if host group has no activation keys.
+    required: false
+    type: list
+    elements: str
+  force:
+    description:
+    - "Clear any previous registration and run C(subscription-manager) with C(--force)."
+    required: false
+    type: bool
+  hostgroup:
+    description:
+    - Host group to register the host in.
+    required: false
+    type: str
+  ignore_subman_errors:
+    description:
+    - Ignore C(subscription-manager) errors for C(subscription-manager register) command.
+    required: false
+    type: bool
+  insecure:
+    description:
+    - Enable insecure argument for the initial C(curl).
+    required: false
+    type: bool
+  jwt_expiration:
+    description:
+    - Expiration of the authorization token (in hours).
```

Comment:
@evgeni @Griffin-Sullivan  , Is it worth mentioning that , the default jwt_expiration period is of 4 hours , just for clarification ?

Reply:
We usually try not to document server side defaults, as those can change and then our docs get out of sync.

Reply:
I was thinking from the point of view of a user who might wonder what is the default expiration period of the token after generating it. But I understand your point as well.  Thnx

Reply:
What is the usual user flow for that function? I mean nobody will install Foreman, fire up an editor and start writing an Ansible playbook to register a host. They will visit the UI, read the docs, etc. So these places should already tell the defaults?

Reply:
True. They should do that only but there are users who need spoonfeeding as well.

The only place from where they can guess the default is `Hosts --> Register Hosts --> Advanced tab` and we pre-populate the `Token life time` field with value 4.  Other than that whether it's API or hammer ,or docs , nowhere we explicitly mention that 4 hours would be the default lifetime. The only thing we mention is `Expiration of the authorization token (in hours)`.

Reply:
Huh, interesting.
Maybe only that UI enforces the 4h and otherwise you get an unlimited token? That's something to investigate, but not on a Sunday morning ;)

Reply:
But as i mentioned earlier, I understand your point of view as well. The default can change and we may not be able to keep the new change in sync with every other place where it needs to be.

Reply:
It's 4 for sure. https://github.com/theforeman/foreman/blob/develop/app/controllers/concerns/foreman/controller/registration_commands.rb#L35-L39

Reply:
Oh huh, it can be the string "unlimited"?! How? The api says the value is a number:
https://github.com/theforeman/foreman/blob/74933559fafc3f52494845bf82a5a67d83b2d584/app/controllers/api/v2/registration_commands_controller.rb#L18

(And we translate this to "int" here in Ansible)

Reply:
That is `kind of` reported in https://bugzilla.redhat.com/show_bug.cgi?id=2180568

Reply:
Interesting, thanks

Reply:
I'd prefer not to diverge from the API, so if we make that accept "unlimited", we should also accept strings here.

But you're right, at the moment non-integers would be rejected

Reply:
https://github.com/theforeman/foreman/pull/9951 - the api is moving to a "string" with stricter validation afterwards, so we should follow.

Reply:
Is it not the opposite i.e. Leos suggested that we move to numbers instead i.e. 

* x < 0 raise an error, invalid value
* x = 0 unlimited expiration
* x > 0 expiration in hours.

and for both UI as well as API.

Reply:
it seems the general consensus in that PR is that the field becomes "number only", with either `0` or `-1` meaning "unlimited".
I think we can go ahead and merge *this* PR as-is (at it already accepts "numbers only"), and once the other PR is in we can adjust the doc string to mention what needs to be done for unlimited?

Reply:
Agreed.. No problems with that I guess.. LGTM ..